### ISSUE TITLE
Xfailed tests that pass will trip the maxfails parameter

### ISSUE THREAD

**Issue by Dude-X** [pytest-dev/pytest#1522]
If you use the `--maxfails=#` in the command line, and some tests are marked as `xfail` if that test passes, then it is treated as a test failure.

This isn't semantically correct since xfails should not fail the test suite unless you explicitly enable `xfail_strict` in py.test.ini.

**Comment by nicoddemus** [comment]
Thanks!

**Comment by RedBeardCode** [comment]
Couldn't reproduce that with the features branch. Tried it with the following example

```
import pytest

@pytest.mark.xfail
def test_1():
	assert True

@pytest.mark.xfail
def test_2():
	assert True

@pytest.mark.xfail
def test_3():
	assert True

@pytest.mark.xfail
def test_4():
	assert True
```

After calling

`py.test --maxfail=2`

I got the following result.

```
test_1522.py XXXX
================================= short test summary info ==================================
XPASS test_1522.py::test_1 
XPASS test_1522.py::test_2 
XPASS test_1522.py::test_3 
XPASS test_1522.py::test_4 

```

So I think this is the right behavior.

**Comment by nicoddemus** [comment]
@RedBeardCode could you please check out `2.9.2` and `2.8.4` and see if you can reproduce then? If so, we can assume it was fixed and can close this, otherwise we should ask for more information. Thanks! :grin:

**Comment by RedBeardCode** [comment]
@nicoddemus I tried it with `2.9.2` and `2.8.4` and got the same result. So it works.

**Comment by nicoddemus** [comment]
Thanks @RedBeardCode! :grin: 

@Dude-X could you post more information for us to reproduce the problem?

**Comment by Dude-X** [comment]
I have in my `pytest.ini` this line: `addopts = --strict`.
I wonder if this strictness for marks is affecting xfail. 

This issue still affects me in CI using PyTest 2.9.2; I usually have to adjust the maxfail option so that my tests don't prematurely bailout, even if they xpass.

**Comment by RonnyPfannschmidt** [comment]
i wonder if preparsing triggers passing strict as strict-xfail, this is related to a python bug that was only fixed in python 3.5 - argparse considers unique prefixes as arguments

**Comment by Zac-HD** [comment]
It's working for me too, so I'm going to close the issue as "not reproducible, probably fixed".
